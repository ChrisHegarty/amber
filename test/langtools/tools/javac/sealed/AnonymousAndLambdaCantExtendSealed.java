/*
 * @test /nodynamiccopyright/
 * @summary smoke test for sealed classes
 * @compile/fail/ref=AnonymousAndLambdaCantExtendSealed.out -XDrawDiagnostics AnonymousAndLambdaCantExtendSealed.java
 */

class AnonymousAndLambdaCantExtendSealed {
    sealed interface I1 extends Runnable {
        public static I1 i = () -> {};
    }

    sealed interface I2 extends Runnable {
        public static void foo() { new I2() { public void run() { } }; }
    }
}
