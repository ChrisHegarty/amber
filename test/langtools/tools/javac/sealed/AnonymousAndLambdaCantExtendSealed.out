AnonymousAndLambdaCantExtendSealed.java:9:30: compiler.err.cant.inherit.from.sealed: AnonymousAndLambdaCantExtendSealed.I1
AnonymousAndLambdaCantExtendSealed.java:13:40: compiler.err.cant.inherit.from.sealed: AnonymousAndLambdaCantExtendSealed.I2
2 errors
